components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -599,6 +606,16 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
                 );
                 return false;
             }
+            // Set commit index for learner snapshots. It's needed to address
+            // compatibility issues between v1 and v2 snapshots.
+            // See https://github.com/pingcap/tiflash/issues/7568#issuecomment-1576382311
+            if let Some(p) = find_peer_by_id(self.region(), to_peer_id) && p.get_role() == PeerRole::Learner  {
```

Comment:
No harm to set it unconditionally?

Reply:
Yes, it's okay. Although I prefer to set it only for learners to make snapshot clearer.